disnake/ui/button.py

```diff
@@ -88,6 +88,17 @@ class Button(Item[V]):
     # We have to set this to MISSING in order to overwrite the abstract property from WrappedComponent
     _underlying: ButtonComponent = MISSING
 
+    @overload
+    def __new__(cls: Type[Button[None]], **kwargs) -> Button[None]:
+        ...
+
+    @overload
+    def __new__(cls: Type[Button[V]], **kwargs) -> Button[V]:
+        ...
+
+    def __new__(cls: Type[Button[Any]], **kwargs) -> Any:
+        return super().__new__(cls)
```

Comment:
These also need to take `*args` as a subclass does not need to use only kwargs.

Reply:
Resolved by switching to overloading `__init__` instead.